sql_schema/Ecommerce.sql

```diff
@@ -0,0 +1,68 @@
+CREATE DATABASE Ecommerce;
+
+USE Ecommerce;
+
+CREATE TABLE Customer(
+CustomerID INT PRIMARY KEY,
+CustomerName VARCHAR(70) NOT NULL,
+CustomerPhone CHAR(10) NOT NULL,
+CustomerEmail VARCHAR(60) NOT NULL,
+CustomerAge INT ,
+CustomerGender CHAR(1));
+
+CREATE TABLE Address(
+AddressID INT PRIMARY KEY,
+CustomerID INT,
+State VARCHAR(20) NOT NULL,
+City VARCHAR(20)NOT NULL,
+Zip VARCHAR(10),
+FOREIGN KEY (CustomerID) REFERENCES Customer(CustomerID));
+
+
+CREATE TABLE Brands(
+BrandID INT PRIMARY KEY,
+BrandName VARCHAR(20) NOT NULL,
+BrandRating FLOAT);
+
+CREATE TABLE Products(
+ProductID INT PRIMARY KEY,
+ProductName VARCHAR(30),
+QuantityAvailable INT NOT NULL,
+ProductDescription VARCHAR(150),
+BrandID INT,
+FOREIGN KEY (BrandID) REFERENCES Brands(BrandID));
+
+CREATE TABLE Review(
+ReviewID INT PRIMARY KEY,
+CustomerID INT,
+ProductID INT,
+Review VARCHAR(300),
+ProductRating FLOAT,
+FOREIGN KEY (CustomerID) REFERENCES Customer(CustomerID),
+FOREIGN KEY (ProductID) REFERENCES Products(ProductID));
+
+CREATE TABLE Orders(
+OrderID INT PRIMARY KEY,
+CustomerID INT,
+ProductID INT,
```

Comment:
How to add multiple products

Reply:
-- Find the constraint names in table orders and copy the constraint name for productID
select CONSTRAINT_NAME
from INFORMATION_SCHEMA.TABLE_CONSTRAINTS
where TABLE_NAME = 'orders';

-- Initialy drop constraint for ProductId foreign key reference and drop column ProductID later in Orders table.

alter table Orders drop constraint FK__Orders__ProductI__45F365D3;
alter table orders drop column productID;

-- Add new table

CREATE TABLE OrderDetails(
OrderDetailID INT PRIMARY KEY,
OrderID INT NOT NULL,
ProductID INT NOT NULL,
Quantity INT NOT NULL,
FOREIGN KEY (OrderID) REFERENCES Orders(OrderID),
FOREIGN KEY (ProductID) REFERENCES Products(ProductID));


-- Find the constraint names for table Payment and copy contraint name for OrderID  

SELECT CONSTRAINT_NAME
FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS
WHERE TABLE_NAME = 'Payment';

-- Initially drop constraint for OrderID foreign key  reference and drop column OrderID later in Payment table.

ALTER TABLE Payment DROP CONSTRAINT FK__Payment__OrderID__48CFD27E;
ALTER TABLE Payment DROP COLUMN OrderID;

-- Add OrderDetailID to Payment table

ALTER TABLE Payment
ADD OrderDetailID INT;

-- Add foreign key reference for OrderDetailID in Payment table
ALTER TABLE Payment
ADD FOREIGN KEY (OrderDetailID) REFERENCES OrderDetails(OrderDetailID);

Reply:
In the new table, we are not storing price ?